apps/studio/src/server/modules/resource/resource.router.ts

```diff
@@ -535,4 +512,36 @@ export const resourceRouter = router({
 
       return ancestors.reverse().slice(0, -1)
     }),
+
+  search: protectedProcedure
```

Comment:
question: can't seem to find it but it seems that we are not validating the user's permissions to view the resources? Would it mean that it is possible for a user with insufficient permissions to be able to view a certain resource that they don't have access to?

Reply:
yup you are right!

decided there's no need for this now as
- users would still be able to see the resources on the left sidebar either way
- the current way of setting resource permission could potentially lead to very nested query to check if a user has permission for each resource

will leave this for later when we want to really to implement a studio-wide resource-based permission

Reply:
Hmm but I thought the permissions is on a site level? It seems like it is possible for someone without any role to access the site to be able to use this search endpoint to get the list of resources for the site, even though by right they don't have access.

Reply:
ah okay i misunderstood what you meant. Will add the check for site-level permission!